src/js/BookReader/Navbar/Navbar.js

```diff
@@ -0,0 +1,180 @@
+    br.refs.$BRnav = this.$nav = $(
+      `<div class="BRnav BRnavDesktop">
+          <div class="BRnavCntl BRnavCntlBtm BRdn js-tooltip" title="Toggle toolbars"></div>
+          ${navTitle ? `<div class="BRnavTitle">${navTitle}</div>` : ''}
+          <div class="BRnavpos">
+            <div class="BRpager"></div>
+            <div class="BRnavline"></div>
+          </div>
+          <div class="BRpage">`
+
+        // Note, it's important for there to not be whitespace
+        + `<span class='BRcurrentpage'></span>`
+        + `<button class="BRicon book_left js-tooltip"></button>`
+        + `<button class="BRicon book_right js-tooltip"></button>`
+        + `<button class="BRicon onepg desktop-only js-tooltip"></button>`
+        + `<button class="BRicon twopg desktop-only js-tooltip"></button>`
+        + `<button class="BRicon thumb desktop-only js-tooltip"></button>`
+        // zoomx
+        + `<button class="BRicon zoom_out desktop-only js-tooltip"></button>`
+        + `<button class="BRicon zoom_in desktop-only js-tooltip"></button>`
+        + `<button class="BRicon full js-tooltip"></button>`
+        + `</div>
+        </div>`);
+  
+    this.$root.append(this.$nav);
+    br.refs.$br.append(this.$root);
+
+    const $slider = this.$root.find('.BRpager').slider({
+      animate: true,
+      min: 0,
+      max: br.getNumLeafs() - 1,
+      value: br.currentIndex(),
+      range: "min"
+    });
+
+    $slider.on('slide', (event, ui) => {
+      this.updateNavPageNum(ui.value);
+      return true;
+    });
+
+    $slider.on('slidechange', (event, ui) => {
+      this.updateNavPageNum(ui.value);
+      // recursion prevention for jumpToIndex
+      if ( $slider.data('swallowchange') ) {
+        $slider.data('swallowchange', false);
+      } else {
+        br.jumpToIndex(ui.value);
+      }
+      return true;
+    });
+
+    this.updateNavPageNum(br.currentIndex());
+
+    return this.$nav;
+  }
+
+  /**
+   * Initialize the navigation bar when embedded
+   */
+  initEmbed() {
+    const { br } = this;
+    // IA-specific
+    let thisLink = (window.location + '')
+      .replace('?ui=embed','')
+      .replace('/stream/', '/details/')
+      .replace('#', '/');
+    const logoHtml = br.showLogo ? `<a class="logo" href="${br.logoURL}" target="_blank"></a>` : '';
+
+    br.refs.$BRfooter = this.$root = $('<div class="BRfooter"></div>');
+    br.refs.$BRnav = this.$nav = $(
+      `<div class="BRnav BRnavEmbed">
+          ${logoHtml}
+          <span class="BRembedreturn">
+             <a href="${thisLink}" target="_blank">${br.bookTitle}</a>
+          </span>
+          <span class="BRtoolbarbuttons">
+            <button class="BRicon book_left"></button>
+            <button class="BRicon book_right"></button>
+            <button class="BRicon full"></button>
+          </span>
+      </div>`);
+    this.$root.append(this.$nav);
+    br.refs.$br.append(this.$root);
```

Comment:
do both navs get appended at both places?

Reply:
Yep! Only one of these `init`s is expected to be run.